### PR TITLE
support gpu to the grid client

### PR DESCRIPTION
**Pull request by AhmedHanafy725**
### Description

support gpu in grid client

### Related Issues

- #464 

### Example

deployment result

```ts
[
  {
    version: 0,
    contractId: 31821,
    nodeId: 93,
    name: 'testvm',
    created: 1687430687,
    status: 'ok',
    message: '',
    flist: 'https://hub.grid.tf/tf-official-apps/base:latest.flist',
    publicIP: null,
    planetary: '301:8728:f02b:166:66d5:4750:94e4:9a88',
    interfaces: [ { network: 'wedtest', ip: '10.249.2.2' } ],
    capacity: { cpu: 8, memory: 16384 },
    mounts: [],
    env: { SSH_KEY: '' },
    entrypoint: '/sbin/zinit init',
    metadata: '{"type":"vm","name":"newVMS","projectName":""}',
    description: 'test deploying VMs via ts grid3 client',
    rootfs_size: 0,
    corex: false,
    gpu: [ '0000:0e:00.0/1002/744c' ]
  }
]
```

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Comment by xmonader** [comment]
Looks really great, can we also include an example script please to deploy

**Comment by AhmedHanafy725** [comment]
> Looks really great, can we also include an example script please to deploy?

done added

**Review by zaelgohary — APPROVED** [review]

File: packages/grid_client/src/high_level/machine.ts [hunk 1]
```diff
@@ -131,6 +133,28 @@ class VMHL extends HighLevelBase {
       }
     }
 
+    if (gpu && gpu.length > 0) {
+      const nodeTwinId = await this.nodes.getNodeTwinId(nodeId);
+      const gpuList = await this.rmb.request([nodeTwinId], "zos.gpu.list", "");
+      if (gpuList.length <= 0) {
+        throw Error(`The selected node ${nodeId} doesn't have GPU card`);
+      }
+      for (const g of gpu) {
```
Comment (Mahmoud-Emad) [inline packages/grid_client/src/high_level/machine.ts:142]: gpus

File: packages/grid_client/src/high_level/machine.ts [hunk 2]
```diff
@@ -45,6 +45,7 @@ class VMHL extends HighLevelBase {
     corex = false,
     solutionProviderId: number,
     zlogsOutput?: string,
+    gpu: string[] = [],
```
Comment (Mahmoud-Emad) [inline packages/grid_client/src/high_level/machine.ts:48]: GPUs, this is a list of GPU IDs no?

File: packages/grid_client/src/high_level/kubernetes.ts [hunk 3]
```diff
@@ -100,6 +102,7 @@ class KubernetesHL extends HighLevelBase {
     corex = false,
     solutionProviderId: number,
     zlogsOutput?: string,
+    gpu: string[] = [],
```
Comment (Mahmoud-Emad) [inline packages/grid_client/src/high_level/kubernetes.ts:105]: GPUs, this is a list of GPU IDs no?

File: packages/grid_client/src/high_level/kubernetes.ts [hunk 4]
```diff
@@ -32,6 +32,7 @@ class KubernetesHL extends HighLevelBase {
...
```
Comment (Mahmoud-Emad) [inline packages/grid_client/src/high_level/kubernetes.ts:35]: GPUs, this is a list of GPU IDs no?

File: packages/grid_client/src/primitives/nodes.ts [hunk 5]
```diff
@@ -332,6 +335,8 @@ class Nodes {
       status: "up",
       page: options.page,
       size: options.size,
+      has_gpu: options.hasGPU,
+      rented_by: options.rentedBy,
```
Comment (Mahmoud-Emad) [inline packages/grid_client/src/primitives/nodes.ts:339]: Why underscores?

**Review by Mahmoud-Emad — APPROVED** [review]

File: packages/grid_client/src/primitives/nodes.ts (hunk 5 above)
Comment (AhmedHanafy725) [reply packages/grid_client/src/primitives/nodes.ts:339]: cause this is how the query looks like in the gridproxy

**Review by Mahmoud-Emad — APPROVED** [review]
LGTM, teslam edak ya hanafy

**Pull request merged**